src/main/java/appeng/util/item/FluidList.java

```diff
@@ -0,0 +1,219 @@
+ */
+
+package appeng.util.item;
+
+
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+
+import com.google.common.collect.Lists;
+
+import appeng.api.config.FuzzyMode;
+import appeng.api.storage.data.IAEFluidStack;
+import appeng.api.storage.data.IItemList;
+
+
+public final class FluidList implements IItemList<IAEFluidStack>
+{
+
+	private final Map<IAEFluidStack, IAEFluidStack> records = new HashMap<IAEFluidStack, IAEFluidStack>();
+
+	@Override
+	public void add( IAEFluidStack option )
+	{
+		if( this.isStackNull( option ) )
+		{
+			return;
+		}
+
+		final IAEFluidStack st = this.getFluidRecord( option );
+
+		if( st != null )
+		{
+			st.add( option );
+			return;
+		}
+
+		final IAEFluidStack opt = option.copy();
+
+		this.putFluidRecord( opt );
+	}
+
+	@Override
+	public IAEFluidStack findPrecise( IAEFluidStack i )
+	{
+		if( this.isStackNull( i ) )
+		{
+			return null;
+		}
+
+		final IAEFluidStack is = this.getFluidRecord( i );
+
+		return is;
+	}
+
+	@Override
+	public Collection<IAEFluidStack> findFuzzy( IAEFluidStack filter, FuzzyMode fuzzy )
+	{
+		if( this.isStackNull( filter ) )
+		{
+			return Collections.emptyList();
+		}
+
+		final List<IAEFluidStack> result = Lists.newArrayList();
+
+		if( filter.equals( this ) )
+		{
+			result.add( filter );
+		}
+
+		return result;
+
+	}
+
+	@Override
+	public boolean isEmpty()
+	{
+		return !this.iterator().hasNext();
+	}
+
+	@Override
+	public void addStorage( IAEFluidStack option )
+	{
+		if( this.isStackNull( option ) )
+		{
+			return;
+		}
+
+		final IAEFluidStack st = this.getFluidRecord( option );
+
+		if( st != null )
+		{
+			st.incStackSize( option.getStackSize() );
+			return;
+		}
+
+		final IAEFluidStack opt = option.copy();
+		this.putFluidRecord( opt );
+	}
+
+	/*
+	 * public synchronized void clean() { Iterator<StackType> i = iterator(); while (i.hasNext()) { StackType AEI =
+	 * i.next(); if ( !AEI.isMeaningful() ) i.remove(); } }
+	 */
+
+	@Override
+	public void addCrafting( IAEFluidStack option )
+	{
+		if( this.isStackNull( option ) )
+		{
+			return;
+		}
+
+		final IAEFluidStack st = this.getFluidRecord( option );
+
+		if( st != null )
+		{
+			st.setCraftable( true );
+			return;
+		}
+
+		final IAEFluidStack opt = option.copy();
+		opt.setStackSize( 0 );
+		opt.setCraftable( true );
+
+		this.putFluidRecord( opt );
+	}
+
+	@Override
+	public void addRequestable( IAEFluidStack option )
+	{
+		if( this.isStackNull( option ) )
+		{
+			return;
+		}
+
+		final IAEFluidStack st = this.getFluidRecord( option );
+
+		if( st != null )
+		{
+			st.setCountRequestable( st.getCountRequestable() + option.getCountRequestable() );
+			return;
+		}
+
+		final IAEFluidStack opt = option.copy();
+		opt.setStackSize( 0 );
+		opt.setCraftable( false );
+		opt.setCountRequestable( opt.getCountRequestable() );
```

Comment:
No change here, so it really is a potential bug from the beginning. Should most likely be `option` not `opt`